internal/http/update.go

```diff
@@ -36,11 +39,13 @@ func (h updateHandler) Routes(r chi.Router) {
 }
 
 func (h updateHandler) getLatest(w http.ResponseWriter, r *http.Request) {
-	latest := h.service.GetLatestRelease(r.Context())
-	if latest != nil {
-		render.Status(r, http.StatusOK)
-		render.JSON(w, r, latest)
-		return
+	if h.config.CheckForUpdates {
+		latest := h.service.GetLatestRelease(r.Context())
+		if latest != nil {
+			render.Status(r, http.StatusOK)
+			render.JSON(w, r, latest)
+			return
+		}
 	}
 
 	render.NoContent(w, r)
```

Comment:
This likely should return something else, so there's an intelligible error presented to the client if the update fails to run. Maybe an icon under the rabbit with a broken cable, or something like that with a mouse-over message.